notebooks/textbook/Hadamard_Test.ipynb

```diff
@@ -0,0 +1,375 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "1.0\n"
+     ]
+    }
+   ],
+   "source": [
+    "device = LocalSimulator()\n",
+    "task = device.run(ht_circuit, shots=1000)\n",
+    "\n",
+    "probs  = task.result().measurement_probabilities\n",
+    "p_zero = probs.get('0', 0)\n",
+    "\n",
+    "real_part = 2 * p_zero - 1\n",
+    "\n",
+    "print(real_part)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Imaginary Part Estimation\n",
+    "\n",
+    "Braket's `hadamard_test_circuit` supports the real part estimation by default, but we can also estimate the imaginary part by setting the `component` argument to `'imaginary'`. We will use the same controlled unitary operator as before."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "T  : │  0  │  1  │  2  │  3  │  4  │\n",
+      "      ┌───┐ ┌───┐       ┌───┐ ┌───┐ \n",
+      "q0 : ─┤ H ├─┤ S ├───●───┤ H ├─┤ M ├─\n",
+      "      └───┘ └───┘   │   └───┘ └───┘ \n",
+      "                  ┌─┴─┐             \n",
+      "q1 : ─────────────┤ U ├─────────────\n",
+      "                  └───┘             \n",
+      "T  : │  0  │  1  │  2  │  3  │  4  │\n"
+     ]
+    }
+   ],
+   "source": [
+    "ht_circuit = hadamard_test_circuit(Qubit(0), controlled_unitary, component='imaginary')\n",
+    "ht_circuit.measure(0)\n",
+    "print(ht_circuit)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The only difference in the circuit is the addition of a phase shift gate to the first qubit, which allows us to estimate the expected value of the imaginary part."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-0.01200000000000001\n"
+     ]
+    }
+   ],
+   "source": [
+    "device = LocalSimulator()\n",
+    "task = device.run(ht_circuit, shots=1000)\n",
+    "\n",
+    "probs  = task.result().measurement_probabilities\n",
+    "p_zero = probs.get('0', 0)\n",
+    "\n",
+    "imaginary_part = 2 * p_zero - 1\n",
+    "\n",
+    "print(imaginary_part)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run on a QPU \n",
+    "\n",
+    "To run the Hadamard test on a QPU, we replace the LocalSimulator with an AwsDevice. The cost to run this experiment is $0.3 per task and $0.00145 per shot on the IQM Garnet device, that totals $1.75 USD. Because Garnet does not support arbitrary unitary operators, we will directly apply the Pauli-Z gate in this example."
```

Comment:
```suggestion
    "To run the Hadamard test on a QPU, we replace the LocalSimulator with an AwsDevice. The cost to run this experiment is \\\\$0.3 per task and \\\\$0.00145 per shot on the IQM Garnet device, which totals \\\\$1.75 USD. Because Garnet does not support arbitrary unitary operators, we will directly apply the Pauli-Z gate in this example."
```